File: languages/java/app-encryption/src/test/java/com/godaddy/asherah/appencryption/AppEncryptionPartitionTest.java
package com.godaddy.asherah.appencryption;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class AppEncryptionPartitionTest {

    private final static String testPartitionId = "test_partition_id";
    private final static String testSystemId = "test_system_id";
    private final static String testProductId = "test_product_id";

    private AppEncryptionPartition appEncryptionPartition;

    @BeforeEach
    void setUp() {
        appEncryptionPartition = new AppEncryptionPartition(testPartitionId, testSystemId, testProductId);
    }

    @Test
    void testGetPartitionId() {
        assertEquals(testPartitionId, appEncryptionPartition.getPartitionId());
    }

    @Test
    void testGetSystemId() {
        assertEquals(testSystemId, appEncryptionPartition.getSystemId());
    }

    @Test
    void testGetProductId() {
        assertEquals(testProductId, appEncryptionPartition.getProductId());
    }

    @Test
    void testGetSystemKeyId() {
        String systemKeyIdString = "_SK_" + testSystemId + "_" + testProductId;
        assertEquals(systemKeyIdString, appEncryptionPartition.getSystemKeyId());
    }

    @Test
    void testGetIntermediateKeyId() {
        String intermediateKeyIdString = "_IK_" + testPartitionId + "_" + testSystemId + "_" + testProductId;
        assertEquals(intermediateKeyIdString, appEncryptionPartition.getIntermediateKeyId());
    }

    @Test
    void testToString() {
        String toStringString = appEncryptionPartition.getClass().getSimpleName() + "[partitionId=" + testPartitionId +
                ", systemId=" + testSystemId + ", productId=" + testProductId + "]";
        assertEquals(toStringString, appEncryptionPartition.toString());
    }
}

File: languages/csharp/AppEncryption/AppEncryption.Tests/AppEncryption/AppEncryptionPartitionTest.cs
using Xunit;

namespace GoDaddy.Asherah.AppEncryption.Tests.AppEncryption
{
    [Collection("Logger Fixture collection")]
    public class AppEncryptionPartitionTest
    {
        private const string TestPartitionId = "test_partition_id";
        private const string TestSystemId = "test_system_id";
        private const string TestProductId = "test_product_id";

        private readonly AppEncryptionPartition appEncryptionPartition;

        public AppEncryptionPartitionTest()
        {
            appEncryptionPartition =
                new AppEncryptionPartition(TestPartitionId, TestSystemId, TestProductId);
        }

        [Fact]
        private void TestPartitionCreation()
        {
            Assert.NotNull(appEncryptionPartition);
        }

        [Fact]
        private void TestGetPartitionId()
        {
            string actualTestPartitionId = appEncryptionPartition.PartitionId;
            Assert.Equal(TestPartitionId, actualTestPartitionId);
        }

        [Fact]
        private void TestGetSystemId()
        {
            string actualSystemId = appEncryptionPartition.SystemId;
            Assert.Equal(TestSystemId, actualSystemId);
        }

        [Fact]
        private void TestGetProductId()
        {
            string actualProductId = appEncryptionPartition.ProductId;
            Assert.Equal(TestProductId, actualProductId);
        }

        [Fact]
        private void TestGetSystemKeyId()
        {
            const string systemKeyIdString = "_SK_" + TestSystemId + "_" + TestProductId;
            Assert.Equal(systemKeyIdString, appEncryptionPartition.SystemKeyId);
        }

        [Fact]
        private void TestGetIntermediateKeyId()
        {
            const string intermediateKeyIdString = "_IK_" + TestPartitionId + "_" + TestSystemId + "_" + TestProductId;
            Assert.Equal(intermediateKeyIdString, appEncryptionPartition.IntermediateKeyId);
        }

        [Fact]
        private void TestToString()
        {
            string expectedToStringString = appEncryptionPartition.GetType().Name + "[partitionId=" + TestPartitionId +
                    ", systemId=" + TestSystemId + ", productId=" + TestProductId + "]";
            Assert.Equal(expectedToStringString, appEncryptionPartition.ToString());
        }
    }
}
